Assets/Plugins/Source/Core/Config.cs

```diff
@@ -37,6 +41,73 @@ namespace PlayEveryWare.EpicOnlineServices
     [Serializable]
     public abstract class Config : ICloneable
     {
+        protected readonly string Filename;
+        protected readonly string Directory;
+
+        private string _lastReadJsonString;
+
+        protected Config(string filename) : this(filename, Path.Combine(Application.streamingAssetsPath, "EOS")) { }
+
+        protected Config(string filename, string directory)
```

Comment:
Make sure that deriving classes do not have publicly accessible constructors, so that they are required to make use of the templated async Get function.

Reply:
This pattern is actually not possible to implement without using reflection.